ifopt/CMakeLists.txt

```diff
@@ -1,4 +1,63 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(ifopt)
-find_package(catkin REQUIRED)
-catkin_metapackage()
+option(CATKIN_BUILD "CATKIN_BUILD" ON)
+MESSAGE(STATUS "CATKIN_BUILD is " ${CATKIN_BUILD})
+
+if(${CATKIN_BUILD})
+    find_package(catkin REQUIRED)
+    catkin_metapackage()
+else()
+    # do native cmake build
+    # add gtest and gmock support
+    # - see http://www.kaizou.org/2014/11/gtest-cmake/
+    # ------------- BEGIN FROM BLOG ------------------
```

Comment:
This whole section for unit testing with gtest seems a bit long. I would prefer to do something like they did here (create and download manually, cmake only checks if folder exists,...):
https://github.com/ethz-asl/kindr
They also nicely use .cmake and find scripts, which I think is a good guidline.